### PR TITLE
Issue 684/person fields

### LINKED ISSUE #684: Display all fields, including custom, on person page
## Description
A user should be able to see all the data from a person object on their profile page.

## Relevant Job Stories
Missing

## Prerequisites
None

## Requirements
- [ ] All fields, including custom fields, should be displayed on person page
- [ ] The "Edit" button should read "Edit fields" or similar
- [ ] Clicking the edit button should navigate to Gen2

## Possible implementations
All fields are available on the person object (may contain more than what are in the TS type, because of custom fields).

Labels for custom field can be found by fetching /orgs/{orgId}/people/fields from the API.

The URL on Gen2 is organize.zetk.in/people/person:<personId>?org=<orgId>

## Design specifications
None

## Open questions
None

### PR DESCRIPTION
**Pull request by djbusstop**
## Description
This PR changes the person details card to show more fields including custom fields.

## Screenshots
Screenshot from 2022-05-30 10-42-58

## Changes

* Adds
  * Redirect from `.../people/[personId]/edit` to the gen2 person page.
  * `personFieldsResource` for fetching all custom fields in an organization
  * Storybook story for the `PersonDetailsCard` 
* Changes
  * Removes eslint requirement for camel case on yml keys. This is because the keys come directly from the server.
  * Refactors `PersonDetailsCard`
    * Shows a select list of native person fields
    * Shows all custom fields that are NOT json
    * Formats URL and Date custom fields
    * Formats the phone and email fields and makes them in to links which open the correct respective app
    * Edit button now in the section header but it redirects to gen2
  * Improves `ZetkinPerson` types so they are now correct and adds `ZetkinCustomField`  

## Related issues
Resolves #684

### CODE REVIEW

File: src/pages/organize/[orgId]/people/[personId]/index.tsx [hunk 1]
```diff
@@ -13,12 +13,16 @@ import SinglePersonLayout from 'layout/organize/SinglePersonLayout';
...
+  personFieldsResource,
+  personResource,
+  personTagsResource,
+} from 'api/people';
 import { scaffold, ScaffoldedGetServerSideProps } from 'utils/next';
 
 export const scaffoldOptions = {
   authLevelRequired: 2,
-  localeScope: ['layout.organize', 'pages.people'],
+  localeScope: ['layout.organize', 'pages.people', 'misc.person'],
```
Comment (richardolsson) [inline src/pages/organize/[orgId]/people/[personId]/index.tsx:25]: I don't think this is necessary. The entire `misc` scope should be automatically included on all pages.

File: .eslintrc.js [hunk 2]
```diff
@@ -93,7 +93,6 @@ module.exports = {
       parser: 'yaml-eslint-parser',
       rules: {
         'yml/sort-keys': 'error',
-        'yml/key-name-casing': ['error', { camelCase: true }],
```
Comment (richardolsson) [inline .eslintrc.js:96]: I'm feeling a little bit anxious about this. I like that we require camel-case and the only situation where I think it's reasonable to deviate from that pattern is when API fields are used directly in localization, in which case there can be an exception in those (rare) files with clear intent.

File: src/components/organize/people/PersonDetailsCard.tsx [hunk 3]
```diff
@@ -1,46 +1,116 @@
...
+import ZetkinSection from 'components/ZetkinSection';
+import { ZetkinCustomField, ZetkinPerson } from 'types/zetkin';
+
+const PersonDetailLink: React.FunctionComponent<{
+  href: string;
+}> = ({ children, href }) => (
+  <NextLink href={href} passHref>
+    <Link>{children}</Link>
+  </NextLink>
+);
```
Comment (richardolsson) [inline src/components/organize/people/PersonDetailsCard.tsx:18]: Is this something that should (eventually) be replaced by a general-purpose `<ZetkinLink>` component?

**Review by richardolsson — DISMISSED** [review]
I like this! I think it's good to merge. I think the lack of tests is fine because there is a nice storybook story instead.

My personal preference is to have stricter linting on the YAML files, but I think I'm open to be convinced (since I'm approving this). 😊

File: src/components/organize/people/PersonDetailsCard.tsx (hunk 3 above)
Comment (djbusstop) [reply src/components/organize/people/PersonDetailsCard.tsx:18]: Maybe/probably, that's something we probably want to do when auditing / implementing a design system

**Comment by djbusstop** [comment]
I left the linting and added `eslint-disable` at the top. One downside of this is that this won't persist when generating the yml files from the translator app. That's why I solved it the way I did. But I decided to change over because I also don't want to remove the rule really. I guess I could put the ignore in the eslint config

**Comment by djbusstop** [comment]
I looked elsewhere in the code and we use an eslint disable on other files, so I think it's good to just use that solution for now.

**Comment by richardolsson** [comment]
> I left the linting and added `eslint-disable` at the top. One downside of this is that this won't persist when generating the yml files from the translator app.

Good point! Let's think about that when we return to the translator app, which I think we should do after the summer. It's been serving us very well on the supporter app project, but we always knew there were some improvements (e.g. PR generation, syntax checking, and now maybe automatically adding eslint exceptions?) that we could try to get done before we start asking volunteers to translate this project.

**Review by richardolsson — APPROVED** [review]
All good! Let's merge. 🚀

**Pull request merged**